### PR TITLE
feat: transferLeads leadStatusCategory transfer

### PR DESCRIPTION
**Pull request by LoisChen68**

### CODE REVIEW

**Comment by LoisChen68** [comment]
[【名單撥打】收藏清單顯示問題](https://trello.com/c/xabkquei/5403-%E3%80%90%E5%90%8D%E5%96%AE%E6%92%A5%E6%89%93%E3%80%91%E6%94%B6%E8%97%8F%E6%B8%85%E5%96%AE%E9%A1%AF%E7%A4%BA%E5%95%8F%E9%A1%8C)

File: src/components/sale/TransferModal.tsx [hunk 1]
```diff
@@ -14,17 +14,94 @@ import {
...
+      alert(`移轉名單有誤: \n${alertMes}`)
+
+      return
+    }
+
+    if (matchedMembers.length > 0) {
+      try {
+        let leadStatusCategoryId = selectedLeadStatusCategoryId ? leadStatusCategories[0]?.id : null
+
+        if (!!selectedLeadStatusCategoryId && !leadStatusCategoryId) {
```
Comment (Ponchimeow) [inline src/components/sale/TransferModal.tsx:70]: `!!selectedLeadStatusCategoryId` could use `Boolean(selectedLeadStatusCategoryId)`

File: src/components/sale/TransferModal.tsx [hunk 2]
```diff
@@ -14,17 +14,94 @@ import {
...
+  const handleTransfer = async (managerId: string) => {
+    if (managerId === '') return
+
+    for (const member of selectedRowLeads) {
+      if (member.appId === appId) {
+        matchedMembers.push(member)
+      } else {
+        nonMatchingMembers.push(member)
+      }
+    }
```
Comment (Ponchimeow) [inline src/components/sale/TransferModal.tsx:55]: `useTransferManagers` 
Hasura already has permissions restricted.

**Review by Ponchimeow — APPROVED** [review]
The work item is HighWay
The comment is unrelated to this feature and can be fixed separately.

**Pull request merged**